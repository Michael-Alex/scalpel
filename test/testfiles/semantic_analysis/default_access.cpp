struct super
{
};

struct s: super
{
	int i;
};

class c: super
{
	int i;
};

