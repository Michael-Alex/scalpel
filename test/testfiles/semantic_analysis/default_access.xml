<namespace id="n0">
 <class name="super" id="c0">
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <class name="s" id="c1">
  <base_classes>
   <base_class id="c0" access="public"/>
  </base_classes>
  <destructor access="public" defined="false">
  </destructor>
  <variable name="i" access="public">
   <type>
    <fundamental_type type="int"/>
   </type>
  </variable>
 </class>
 <class name="c" id="c2">
  <base_classes>
   <base_class id="c0" access="private"/>
  </base_classes>
  <destructor access="public" defined="false">
  </destructor>
  <variable name="i" access="private">
   <type>
    <fundamental_type type="int"/>
   </type>
  </variable>
 </class>
</namespace>
