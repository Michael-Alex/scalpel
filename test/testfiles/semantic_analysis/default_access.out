{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "base",
      "destructor":
      {
        "access": "public"
      }
    },
    {
      "id": 1,
      "name": "c",
      "base classes":
      [
        {
          "class id": 0,
          "access": "private"
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": 0,
          "name": "i",
          "access": "private",
          "type":
          {
            "fundamental type": "int"
          }
        }
      ]
    },
    {
      "id": 2,
      "name": "s",
      "base classes":
      [
        {
          "class id": 0,
          "access": "public"
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": 1,
          "name": "i",
          "access": "public",
          "type":
          {
            "fundamental type": "int"
          }
        }
      ]
    }
  ]
}
